### PR TITLE
Adds catching where it was missing and stores failed information for workflow in redis

### PR DESCRIPTION
**Pull request by gabzim**
Accomplishes 2 this:
1. Stores error information in redis when a workflow fails. To remove later: stacktrace. Once we have structured logging in place.
2. Added a catch for when fetching missing vaas fails so it doesn't interrupt the spy workflow.

### CODE REVIEW

File: relayer-engine/src/listener/missedVaaFetching.ts [hunk 1]
```diff
@@ -136,6 +144,8 @@ async function tryFetchAndConsumeVaa(
       emitterAddress,
       sequence.toString(),
       { transport: grpcWebNodeHttpTransport.NodeHttpTransport() },
+      3000,
+      2,
```
Comment (gabzim) [inline relayer-engine/src/listener/missedVaaFetching.ts:148]: 3 second timeout to query for VAA. 2 max retries. 

Should we make this into a setting? I hardcoded to follow the policy of don't make it over-configurable but maybe there's merit here?

**Comment by gabzim** [comment]
closes #62

**Review by sebastianscatularo — APPROVED** [review]

**Review by JoeHowarth — CHANGES_REQUESTED** [review]

File: relayer-engine/src/listener/missedVaaFetching.ts (hunk 1 above)
Comment (gabzim) [reply relayer-engine/src/listener/missedVaaFetching.ts:148]: ooof that is a bug in my opinion, the delay should be at the end of the loop.

File: relayer-engine/packages/relayer-plugin-interface/src/index.ts [hunk 2]
```diff
@@ -47,6 +47,8 @@ export interface Workflow<D = any> {
   maxRetries?: number;
   data: D;
   failedAt?: Date;
+  errorMessage?: string;
```
[the remark this answers is not in the document]
Comment (gabzim) [reply relayer-engine/packages/relayer-plugin-interface/src/index.ts:50]: the plugin author receives the workflow, I could pull the errorMessage out of the workflow and only use it in storage, but, if in the future they want their plugin to handle dead letter queue workflows or do special logic based on what failed, it's going to be hard without exposing the error to them.

[the remark this answers is not in the document]
Comment (JoeHowarth) [reply relayer-engine/packages/relayer-plugin-interface/src/index.ts:50]: Yeah this was a meta comment about storage fields not being on the plugin api. We should probably have 
```
StorageWorkflow {
    Workflow {
         WorkflowData: any
    }
}
```
where only workflow and workflow data are exposed to the plugin. Basically it feels like a code smell that the internal storage format is part of the public api

**Review by JoeHowarth — APPROVED** [review]

**Pull request merged**